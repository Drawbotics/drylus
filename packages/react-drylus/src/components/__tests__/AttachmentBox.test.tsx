import React from 'react';
import { create } from 'react-test-renderer';

import { AttachmentBox } from '../AttachmentBox';

const simplestAttachment = {
  id: '1',
  fileName: 'FirstFile.jpg',
};

const attachmentWithProgress = {
  id: '2',
  fileName: 'ProgressFile.jpg',
  progress: 0.5,
};

const completedAttachment = {
  id: '3',
  fileName: 'CompleteFile.jpg',
  progress: 1,
  url: 'https://www.aws.com',
};

describe('AttachmentBox', () => {
  describe('matches snapshot when', () => {
    it('is given the simplest attachment', () => {
      const tree = create(<AttachmentBox attachment={simplestAttachment} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('is given attachment with progress', () => {
      const tree = create(<AttachmentBox attachment={attachmentWithProgress} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('is given a completed attachment', () => {
      const tree = create(<AttachmentBox attachment={completedAttachment} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('can download attachment', () => {
      const tree = create(
        <AttachmentBox attachment={completedAttachment} onClickDownload={() => {}} />,
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('can remove attachment', () => {
      const tree = create(
        <AttachmentBox attachment={simplestAttachment} onClickClose={() => {}} />,
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
