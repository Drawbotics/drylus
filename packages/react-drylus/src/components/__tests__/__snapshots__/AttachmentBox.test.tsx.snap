// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AttachmentBox matches snapshot when can download attachment 1`] = `
.emotion-7 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.emotion-17 {
  background: #F1F5F9;
  border-radius: 4px;
  color: #172b4e;
  height: 32px;
  padding: 0px 4px;
}

.emotion-16 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-0 {
  display: inline-block;
  position: relative;
  border: 1.2px solid #8C8C8C;
  border-image-slice: 1;
  width: 16px;
  height: 24px;
  margin-left: calc(8px - 1px);
  background: linear-gradient( 45deg, #fff calc(100% - 5px), rgba(255,255,255,0) calc(100% - 5px), rgba(255,255,255,0) 100% );
  -webkit-clip-path: polygon( calc(100% - 7.5px) 0, 100% 7.5px, 100% 100%, -8px 100%, -8px 0 );
  clip-path: polygon( calc(100% - 7.5px) 0, 100% 7.5px, 100% 100%, -8px 100%, -8px 0 );
}

.emotion-0::before {
  content: ' ';
  position: absolute;
  top: -1px;
  right: -1px;
  height: 8px;
  width: 8px;
  box-shadow: -1px 1px 0px 0.2px #8C8C8C;
  border-bottom-left-radius: 1px;
  background: linear-gradient( 45deg, #fff calc(50% - 0.5px), #8C8C8C calc(50% - 0.5px), #8C8C8C calc(50% + 0.5px), transparent calc(50% + 0.5px), transparent 100% );
}

.emotion-0::after {
  content: ' ';
  position: absolute;
  height: calc(8px + 1px);
  width: calc(16px + 2px);
  bottom: 3px;
  left: calc(8px * -1);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-transform: uppercase;
  font-size: 7px;
  color: #fff;
  background: #8C8C8C;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  padding-top: 1px;
  padding-left: 1px;
  font-family: "Rubik",sans-serif;
  font-weight: 400;
}

.emotion-0[data-type]::after {
  content: attr(data-type);
}

.emotion-0::after {
  background: #2A41A2;
}

.emotion-2 {
  margin: 24px;
  margin: 0;
  margin-right: 8px;
}

@media only screen and (max-width:768px) {
  .emotion-2 {
    margin: 16px;
  }
}

@media only screen and (max-width:768px) {
  .emotion-2 {
    margin: 0;
  }
}

@media only screen and (max-width:768px) {
  .emotion-2 {
    margin-right: calc(8px / 2);
  }
}

.emotion-4 {
  display: inline;
  color: rgba(135,155,166,0.5);
  font-size: 0.9rem;
}

.emotion-6 {
  height: 8px;
  width: 100%;
  position: relative;
  border-radius: 8px;
  overflow: hidden;
  background: #eaeff4;
  height: calc(8px / 2);
  border-radius: calc(8px / 2);
}

.emotion-6 [data-element='bar'] {
  border-radius: 8px;
}

.emotion-6 [data-element='bar'] {
  border-radius: calc(8px / 2);
}

.emotion-5 {
  position: absolute;
  z-index: 2;
  top: 0;
  left: 0;
  height: 100%;
  width: 0;
  background: #879ba6;
  -webkit-transition: all 0.15s ease-in-out;
  transition: all 0.15s ease-in-out;
  overflow: hidden;
  background: #3FBC9B;
}

.emotion-5::after {
  background: #879ba6;
  border-radius: 0 !important;
  background-image: linear-gradient( to right, #879ba6 8%, #d3dce6 18%, #879ba6 33% ) !important;
}

.emotion-5::after {
  background: #3FBC9B;
  background-image: linear-gradient( to right, #3FBC9B 8%, #C1E8DE 18%, #3FBC9B 33% ) !important;
}

.emotion-10 {
  margin: 24px;
  margin: 0;
  margin-right: 16px;
}

@media only screen and (max-width:768px) {
  .emotion-10 {
    margin: 16px;
  }
}

@media only screen and (max-width:768px) {
  .emotion-10 {
    margin: 0;
  }
}

@media only screen and (max-width:768px) {
  .emotion-10 {
    margin-right: 8px;
  }
}

.emotion-12 {
  color: inherit;
}

.emotion-12:hover {
  cursor: pointer;
}

<div
  className="emotion-17"
>
  <div
    className="emotion-16"
    style={
      Object {
        "height": "100%",
      }
    }
  >
    <div
      className="emotion-7"
      style={
        Object {
          "display": "flex",
          "minWidth": 0,
        }
      }
    >
      <div
        className="emotion-8"
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <div
          className="emotion-1"
          style={
            Object {
              "display": "flex",
            }
          }
        >
          <div
            className="emotion-0"
            data-type="jpg"
          />
        </div>
        <div
          className="emotion-1"
          style={Object {}}
        >
          <div
            className="emotion-2"
          />
        </div>
        <div
          className="emotion-7"
          style={
            Object {
              "minWidth": 0,
            }
          }
        >
          <div
            style={
              Object {
                "overflow": "hidden",
                "textOverflow": "ellipsis",
                "whiteSpace": "nowrap",
              }
            }
          >
            <span
              className="emotion-4"
            >
              CompleteFile.jpg
            </span>
          </div>
          <div
            className="emotion-6"
            style={
              Object {
                "backgroundColor": "#d3dce6",
                "marginTop": 4,
              }
            }
          >
            <div
              className="emotion-5"
              data-element="bar"
              style={
                Object {
                  "width": "100%",
                }
              }
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="emotion-1"
      style={Object {}}
    >
      <div
        className="emotion-10"
      />
    </div>
    <div
      className="emotion-1"
      style={
        Object {
          "display": "flex",
        }
      }
    >
      <i
        className="Drycon Drycon-download emotion-12"
        onClick={[Function]}
      />
    </div>
    <div
      className="emotion-1"
      style={Object {}}
    >
      <div
        className="emotion-2"
      />
    </div>
  </div>
</div>
`;

exports[`AttachmentBox matches snapshot when can remove attachment 1`] = `
.emotion-5 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.emotion-13 {
  background: #F1F5F9;
  border-radius: 4px;
  color: #172b4e;
  height: 32px;
  padding: 0px 4px;
}

.emotion-12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-0 {
  display: inline-block;
  position: relative;
  border: 1.2px solid #8C8C8C;
  border-image-slice: 1;
  width: 16px;
  height: 24px;
  margin-left: calc(8px - 1px);
  background: linear-gradient( 45deg, #fff calc(100% - 5px), rgba(255,255,255,0) calc(100% - 5px), rgba(255,255,255,0) 100% );
  -webkit-clip-path: polygon( calc(100% - 7.5px) 0, 100% 7.5px, 100% 100%, -8px 100%, -8px 0 );
  clip-path: polygon( calc(100% - 7.5px) 0, 100% 7.5px, 100% 100%, -8px 100%, -8px 0 );
}

.emotion-0::before {
  content: ' ';
  position: absolute;
  top: -1px;
  right: -1px;
  height: 8px;
  width: 8px;
  box-shadow: -1px 1px 0px 0.2px #8C8C8C;
  border-bottom-left-radius: 1px;
  background: linear-gradient( 45deg, #fff calc(50% - 0.5px), #8C8C8C calc(50% - 0.5px), #8C8C8C calc(50% + 0.5px), transparent calc(50% + 0.5px), transparent 100% );
}

.emotion-0::after {
  content: ' ';
  position: absolute;
  height: calc(8px + 1px);
  width: calc(16px + 2px);
  bottom: 3px;
  left: calc(8px * -1);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-transform: uppercase;
  font-size: 7px;
  color: #fff;
  background: #8C8C8C;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  padding-top: 1px;
  padding-left: 1px;
  font-family: "Rubik",sans-serif;
  font-weight: 400;
}

.emotion-0[data-type]::after {
  content: attr(data-type);
}

.emotion-0::after {
  background: #2A41A2;
}

.emotion-2 {
  margin: 24px;
  margin: 0;
  margin-right: 8px;
}

@media only screen and (max-width:768px) {
  .emotion-2 {
    margin: 16px;
  }
}

@media only screen and (max-width:768px) {
  .emotion-2 {
    margin: 0;
  }
}

@media only screen and (max-width:768px) {
  .emotion-2 {
    margin-right: calc(8px / 2);
  }
}

.emotion-4 {
  display: inline;
  color: #172b4e;
  font-size: 1rem;
}

.emotion-10 {
  color: inherit;
}

.emotion-10:hover {
  cursor: pointer;
}

<div
  className="emotion-13"
>
  <div
    className="emotion-12"
    style={
      Object {
        "height": "100%",
      }
    }
  >
    <div
      className="emotion-5"
      style={
        Object {
          "display": "flex",
          "minWidth": 0,
        }
      }
    >
      <div
        className="emotion-6"
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <div
          className="emotion-1"
          style={
            Object {
              "display": "flex",
            }
          }
        >
          <div
            className="emotion-0"
            data-type="jpg"
          />
        </div>
        <div
          className="emotion-1"
          style={Object {}}
        >
          <div
            className="emotion-2"
          />
        </div>
        <div
          className="emotion-5"
          style={
            Object {
              "minWidth": 0,
            }
          }
        >
          <div
            style={
              Object {
                "overflow": "hidden",
                "textOverflow": "ellipsis",
                "whiteSpace": "nowrap",
              }
            }
          >
            <span
              className="emotion-4"
            >
              FirstFile.jpg
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="emotion-1"
      style={Object {}}
    >
      <div
        className="emotion-2"
      />
    </div>
    <div
      className="emotion-1"
      style={
        Object {
          "display": "flex",
        }
      }
    >
      <i
        className="Drycon Drycon-x emotion-10"
        onClick={[Function]}
      />
    </div>
  </div>
</div>
`;

exports[`AttachmentBox matches snapshot when is given a completed attachment 1`] = `
.emotion-7 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.emotion-13 {
  background: #F1F5F9;
  border-radius: 4px;
  color: #172b4e;
  height: 32px;
  padding: 0px 4px;
}

.emotion-12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-0 {
  display: inline-block;
  position: relative;
  border: 1.2px solid #8C8C8C;
  border-image-slice: 1;
  width: 16px;
  height: 24px;
  margin-left: calc(8px - 1px);
  background: linear-gradient( 45deg, #fff calc(100% - 5px), rgba(255,255,255,0) calc(100% - 5px), rgba(255,255,255,0) 100% );
  -webkit-clip-path: polygon( calc(100% - 7.5px) 0, 100% 7.5px, 100% 100%, -8px 100%, -8px 0 );
  clip-path: polygon( calc(100% - 7.5px) 0, 100% 7.5px, 100% 100%, -8px 100%, -8px 0 );
}

.emotion-0::before {
  content: ' ';
  position: absolute;
  top: -1px;
  right: -1px;
  height: 8px;
  width: 8px;
  box-shadow: -1px 1px 0px 0.2px #8C8C8C;
  border-bottom-left-radius: 1px;
  background: linear-gradient( 45deg, #fff calc(50% - 0.5px), #8C8C8C calc(50% - 0.5px), #8C8C8C calc(50% + 0.5px), transparent calc(50% + 0.5px), transparent 100% );
}

.emotion-0::after {
  content: ' ';
  position: absolute;
  height: calc(8px + 1px);
  width: calc(16px + 2px);
  bottom: 3px;
  left: calc(8px * -1);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-transform: uppercase;
  font-size: 7px;
  color: #fff;
  background: #8C8C8C;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  padding-top: 1px;
  padding-left: 1px;
  font-family: "Rubik",sans-serif;
  font-weight: 400;
}

.emotion-0[data-type]::after {
  content: attr(data-type);
}

.emotion-0::after {
  background: #2A41A2;
}

.emotion-2 {
  margin: 24px;
  margin: 0;
  margin-right: 8px;
}

@media only screen and (max-width:768px) {
  .emotion-2 {
    margin: 16px;
  }
}

@media only screen and (max-width:768px) {
  .emotion-2 {
    margin: 0;
  }
}

@media only screen and (max-width:768px) {
  .emotion-2 {
    margin-right: calc(8px / 2);
  }
}

.emotion-4 {
  display: inline;
  color: rgba(135,155,166,0.5);
  font-size: 0.9rem;
}

.emotion-6 {
  height: 8px;
  width: 100%;
  position: relative;
  border-radius: 8px;
  overflow: hidden;
  background: #eaeff4;
  height: calc(8px / 2);
  border-radius: calc(8px / 2);
}

.emotion-6 [data-element='bar'] {
  border-radius: 8px;
}

.emotion-6 [data-element='bar'] {
  border-radius: calc(8px / 2);
}

.emotion-5 {
  position: absolute;
  z-index: 2;
  top: 0;
  left: 0;
  height: 100%;
  width: 0;
  background: #879ba6;
  -webkit-transition: all 0.15s ease-in-out;
  transition: all 0.15s ease-in-out;
  overflow: hidden;
  background: #3FBC9B;
}

.emotion-5::after {
  background: #879ba6;
  border-radius: 0 !important;
  background-image: linear-gradient( to right, #879ba6 8%, #d3dce6 18%, #879ba6 33% ) !important;
}

.emotion-5::after {
  background: #3FBC9B;
  background-image: linear-gradient( to right, #3FBC9B 8%, #C1E8DE 18%, #3FBC9B 33% ) !important;
}

<div
  className="emotion-13"
>
  <div
    className="emotion-12"
    style={
      Object {
        "height": "100%",
      }
    }
  >
    <div
      className="emotion-7"
      style={
        Object {
          "display": "flex",
          "minWidth": 0,
        }
      }
    >
      <div
        className="emotion-8"
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <div
          className="emotion-1"
          style={
            Object {
              "display": "flex",
            }
          }
        >
          <div
            className="emotion-0"
            data-type="jpg"
          />
        </div>
        <div
          className="emotion-1"
          style={Object {}}
        >
          <div
            className="emotion-2"
          />
        </div>
        <div
          className="emotion-7"
          style={
            Object {
              "minWidth": 0,
            }
          }
        >
          <div
            style={
              Object {
                "overflow": "hidden",
                "textOverflow": "ellipsis",
                "whiteSpace": "nowrap",
              }
            }
          >
            <span
              className="emotion-4"
            >
              CompleteFile.jpg
            </span>
          </div>
          <div
            className="emotion-6"
            style={
              Object {
                "backgroundColor": "#d3dce6",
                "marginTop": 4,
              }
            }
          >
            <div
              className="emotion-5"
              data-element="bar"
              style={
                Object {
                  "width": "100%",
                }
              }
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="emotion-1"
      style={Object {}}
    >
      <div
        className="emotion-2"
      />
    </div>
  </div>
</div>
`;

exports[`AttachmentBox matches snapshot when is given attachment with progress 1`] = `
.emotion-7 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.emotion-13 {
  background: #F1F5F9;
  border-radius: 4px;
  color: #172b4e;
  height: 32px;
  padding: 0px 4px;
}

.emotion-12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-0 {
  display: inline-block;
  position: relative;
  border: 1.2px solid #8C8C8C;
  border-image-slice: 1;
  width: 16px;
  height: 24px;
  margin-left: calc(8px - 1px);
  background: linear-gradient( 45deg, #fff calc(100% - 5px), rgba(255,255,255,0) calc(100% - 5px), rgba(255,255,255,0) 100% );
  -webkit-clip-path: polygon( calc(100% - 7.5px) 0, 100% 7.5px, 100% 100%, -8px 100%, -8px 0 );
  clip-path: polygon( calc(100% - 7.5px) 0, 100% 7.5px, 100% 100%, -8px 100%, -8px 0 );
}

.emotion-0::before {
  content: ' ';
  position: absolute;
  top: -1px;
  right: -1px;
  height: 8px;
  width: 8px;
  box-shadow: -1px 1px 0px 0.2px #8C8C8C;
  border-bottom-left-radius: 1px;
  background: linear-gradient( 45deg, #fff calc(50% - 0.5px), #8C8C8C calc(50% - 0.5px), #8C8C8C calc(50% + 0.5px), transparent calc(50% + 0.5px), transparent 100% );
}

.emotion-0::after {
  content: ' ';
  position: absolute;
  height: calc(8px + 1px);
  width: calc(16px + 2px);
  bottom: 3px;
  left: calc(8px * -1);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-transform: uppercase;
  font-size: 7px;
  color: #fff;
  background: #8C8C8C;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  padding-top: 1px;
  padding-left: 1px;
  font-family: "Rubik",sans-serif;
  font-weight: 400;
}

.emotion-0[data-type]::after {
  content: attr(data-type);
}

.emotion-0::after {
  background: #2A41A2;
}

.emotion-2 {
  margin: 24px;
  margin: 0;
  margin-right: 8px;
}

@media only screen and (max-width:768px) {
  .emotion-2 {
    margin: 16px;
  }
}

@media only screen and (max-width:768px) {
  .emotion-2 {
    margin: 0;
  }
}

@media only screen and (max-width:768px) {
  .emotion-2 {
    margin-right: calc(8px / 2);
  }
}

.emotion-4 {
  display: inline;
  color: rgba(135,155,166,0.5);
  font-size: 0.9rem;
}

.emotion-6 {
  height: 8px;
  width: 100%;
  position: relative;
  border-radius: 8px;
  overflow: hidden;
  background: #eaeff4;
  height: calc(8px / 2);
  border-radius: calc(8px / 2);
}

.emotion-6 [data-element='bar'] {
  border-radius: 8px;
}

.emotion-6 [data-element='bar'] {
  border-radius: calc(8px / 2);
}

.emotion-5 {
  position: absolute;
  z-index: 2;
  top: 0;
  left: 0;
  height: 100%;
  width: 0;
  background: #879ba6;
  -webkit-transition: all 0.15s ease-in-out;
  transition: all 0.15s ease-in-out;
  overflow: hidden;
  background: #3FBC9B;
}

.emotion-5::after {
  background: #879ba6;
  border-radius: 0 !important;
  background-image: linear-gradient( to right, #879ba6 8%, #d3dce6 18%, #879ba6 33% ) !important;
}

.emotion-5::after {
  background: #3FBC9B;
  background-image: linear-gradient( to right, #3FBC9B 8%, #C1E8DE 18%, #3FBC9B 33% ) !important;
}

<div
  className="emotion-13"
>
  <div
    className="emotion-12"
    style={
      Object {
        "height": "100%",
      }
    }
  >
    <div
      className="emotion-7"
      style={
        Object {
          "display": "flex",
          "minWidth": 0,
        }
      }
    >
      <div
        className="emotion-8"
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <div
          className="emotion-1"
          style={
            Object {
              "display": "flex",
            }
          }
        >
          <div
            className="emotion-0"
            data-type="jpg"
          />
        </div>
        <div
          className="emotion-1"
          style={Object {}}
        >
          <div
            className="emotion-2"
          />
        </div>
        <div
          className="emotion-7"
          style={
            Object {
              "minWidth": 0,
            }
          }
        >
          <div
            style={
              Object {
                "overflow": "hidden",
                "textOverflow": "ellipsis",
                "whiteSpace": "nowrap",
              }
            }
          >
            <span
              className="emotion-4"
            >
              ProgressFile.jpg
            </span>
          </div>
          <div
            className="emotion-6"
            style={
              Object {
                "backgroundColor": "#d3dce6",
                "marginTop": 4,
              }
            }
          >
            <div
              className="emotion-5"
              data-element="bar"
              style={
                Object {
                  "width": "50%",
                }
              }
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="emotion-1"
      style={Object {}}
    >
      <div
        className="emotion-2"
      />
    </div>
  </div>
</div>
`;

exports[`AttachmentBox matches snapshot when is given the simplest attachment 1`] = `
.emotion-5 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.emotion-11 {
  background: #F1F5F9;
  border-radius: 4px;
  color: #172b4e;
  height: 32px;
  padding: 0px 4px;
}

.emotion-10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-0 {
  display: inline-block;
  position: relative;
  border: 1.2px solid #8C8C8C;
  border-image-slice: 1;
  width: 16px;
  height: 24px;
  margin-left: calc(8px - 1px);
  background: linear-gradient( 45deg, #fff calc(100% - 5px), rgba(255,255,255,0) calc(100% - 5px), rgba(255,255,255,0) 100% );
  -webkit-clip-path: polygon( calc(100% - 7.5px) 0, 100% 7.5px, 100% 100%, -8px 100%, -8px 0 );
  clip-path: polygon( calc(100% - 7.5px) 0, 100% 7.5px, 100% 100%, -8px 100%, -8px 0 );
}

.emotion-0::before {
  content: ' ';
  position: absolute;
  top: -1px;
  right: -1px;
  height: 8px;
  width: 8px;
  box-shadow: -1px 1px 0px 0.2px #8C8C8C;
  border-bottom-left-radius: 1px;
  background: linear-gradient( 45deg, #fff calc(50% - 0.5px), #8C8C8C calc(50% - 0.5px), #8C8C8C calc(50% + 0.5px), transparent calc(50% + 0.5px), transparent 100% );
}

.emotion-0::after {
  content: ' ';
  position: absolute;
  height: calc(8px + 1px);
  width: calc(16px + 2px);
  bottom: 3px;
  left: calc(8px * -1);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-transform: uppercase;
  font-size: 7px;
  color: #fff;
  background: #8C8C8C;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  padding-top: 1px;
  padding-left: 1px;
  font-family: "Rubik",sans-serif;
  font-weight: 400;
}

.emotion-0[data-type]::after {
  content: attr(data-type);
}

.emotion-0::after {
  background: #2A41A2;
}

.emotion-2 {
  margin: 24px;
  margin: 0;
  margin-right: 8px;
}

@media only screen and (max-width:768px) {
  .emotion-2 {
    margin: 16px;
  }
}

@media only screen and (max-width:768px) {
  .emotion-2 {
    margin: 0;
  }
}

@media only screen and (max-width:768px) {
  .emotion-2 {
    margin-right: calc(8px / 2);
  }
}

.emotion-4 {
  display: inline;
  color: #172b4e;
  font-size: 1rem;
}

<div
  className="emotion-11"
>
  <div
    className="emotion-10"
    style={
      Object {
        "height": "100%",
      }
    }
  >
    <div
      className="emotion-5"
      style={
        Object {
          "display": "flex",
          "minWidth": 0,
        }
      }
    >
      <div
        className="emotion-6"
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <div
          className="emotion-1"
          style={
            Object {
              "display": "flex",
            }
          }
        >
          <div
            className="emotion-0"
            data-type="jpg"
          />
        </div>
        <div
          className="emotion-1"
          style={Object {}}
        >
          <div
            className="emotion-2"
          />
        </div>
        <div
          className="emotion-5"
          style={
            Object {
              "minWidth": 0,
            }
          }
        >
          <div
            style={
              Object {
                "overflow": "hidden",
                "textOverflow": "ellipsis",
                "whiteSpace": "nowrap",
              }
            }
          >
            <span
              className="emotion-4"
            >
              FirstFile.jpg
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="emotion-1"
      style={Object {}}
    >
      <div
        className="emotion-2"
      />
    </div>
  </div>
</div>
`;
